spring-core/src/test/java/org/springframework/util/MultiValueMapRelatedTests.java

```diff
@@ -0,0 +1,214 @@
+ */
+
+package org.springframework.util;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Mihai Dumitrescu
+ * @author Arjen Poutsma
+ * @author Juergen Hoeller
+ */
+class MultiValueMapRelatedTests {
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void add(MultiValueMap<String, String> objectUnderTest) {
+		int startingSize = objectUnderTest.size();
+		objectUnderTest.add("key", "value1");
+		objectUnderTest.add("key", "value2");
+		assertThat(objectUnderTest).hasSize(startingSize + 1);
+		assertThat(objectUnderTest.get("key")).containsExactly("value1", "value2");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void addIfAbsentWhenAbsent(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.addIfAbsent("key", "value1");
+		assertThat(objectUnderTest.get("key")).containsExactly("value1");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void addIfAbsentWhenPresent(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.add("key", "value1");
+		objectUnderTest.addIfAbsent("key", "value2");
+		assertThat(objectUnderTest.get("key")).containsExactly("value1");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void set(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.set("key", "value1");
+		objectUnderTest.set("key", "value2");
+		assertThat(objectUnderTest.get("key")).containsExactly("value2");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void addAll(MultiValueMap<String, String> objectUnderTest) {
+		int  startingSize = objectUnderTest.size();
+
+		objectUnderTest.add("key", "value1");
+		objectUnderTest.addAll("key", Arrays.asList("value2", "value3"));
+		assertThat(objectUnderTest).hasSize(startingSize + 1);
+		assertThat(objectUnderTest.get("key")).containsExactly("value1", "value2", "value3");
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	@Disabled("to be fixed in gh-25140")
+	void addAllWithEmptyList(MultiValueMap<String, String> objectUnderTest) {
+		objectUnderTest.addAll("key", Collections.emptyList());
+		assertThat(objectUnderTest).hasSize(1);
+		assertThat(objectUnderTest.get("key")).isEmpty();
+		assertThat(objectUnderTest.getFirst("key")).isNull();
+	}
+
+	@ParameterizedTest
+	@MethodSource("objectsUnderTest")
+	void getFirst(MultiValueMap<String, String> objectUnderTest) {
+		List<String> values = new ArrayList<>(2);
+		values.add("value1");
+		values.add("value2");
+		objectUnderTest.put("key", values);
```

Comment:
We tend to use `Arrays.asList` for such use cases.
```suggestion
		objectUnderTest.put("key", Arrays.asList("value1", "value2"));
```